packages/relayer/message/wait_header_synced.go

```diff
@@ -0,0 +1,55 @@
+package message
+
+import (
+	"context"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/taikoxyz/taiko-mono/packages/relayer/contracts"
+)
+
+func (p *Processor) waitHeaderSynced(ctx context.Context, event *contracts.BridgeMessageSent) error {
+	ticker := time.NewTicker(time.Duration(p.headerSyncIntervalSeconds) * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-ticker.C:
+			// get latest synced header since not every header is synced from L1 => L2,
+			// and later blocks still have the storage trie proof from previous blocks.
+			latestSyncedHeader, err := p.destHeaderSyncer.GetLatestSyncedHeader(&bind.CallOpts{})
+			if err != nil {
+				return errors.Wrap(err, "p.destHeaderSyncer.GetLatestSyncedHeader")
+			}
+
+			block, err := p.srcEthClient.BlockByHash(ctx, latestSyncedHeader)
```

Comment:
Seems here we only need the block header, but not the transactions inside, and since this ticker maybe called multiple times, I think we can use `HeaderByHash` instead of `BlockByHash` to save one request each tick:
![image](https://user-images.githubusercontent.com/104078303/207752042-f0b59df0-8c76-402e-a982-8f7019949506.png)